app.rb

```diff
@@ -28,6 +30,12 @@ def auth_server_uri
     @auth_server_uri ||= URI('https://localhost:1234')
   end
 
+  # Removes all non-alphanumeric characters, leaves in dashes
+  # Useful for extracting slugs from params
+  def sanitize(param)
+    param.to_s.gsub(/[^0-9a-zA-Z-]/, '').presence
```

Comment:
When is `.presence` relevant here?

Reply:
So the way this is used is `sanitize(params[:ial]) || '1'` so since in Ruby empty strings are truthy, this `.presence` has this return `nil` for an empty value and the `|| '1'` keeps on working